docs: add databend example (#1)
ZhiHanZ/indexedxyz-docs: docs/examples/databend.md

---
title: Databend Cloud
---

In this document we'll show how to set up a simple data pipeline to load data from indexed.xyz to databend cloud and perform data analysis through SQL.

Currently we are under Beta Access and You could easily take a trial on databend cloud thourgh the [link](https://app.databend.com/)

## Load data from indexed.xyz

> 

Reviewer: Better to add a explaination: such as all the queries are running in the worksheet, and worksheet doc is in https://docs.databend.com/using-databend-cloud/worksheet

Author: Added explaination:
We would use databend [worksheet](https://docs.databend.com/using-databend-cloud/worksheet) to run all queries with the XSmall instance.

Currently all files stored in indexed.xyz in under [cloudflare R2](https://www.cloudflare.com/products/r2/), and we are using prefix `9d` for the demo.

> At First, we need to add indexed.xyz R2 as a databend [stage](https://databend.rs/doc/sql-commands/ddl/stage/).

Reviewer: ```suggestion
At First, we need to add indexed.xyz R2 as a databend [stage](https://docs.databend.com/sql/sql-commands/ddl/stage/).
```

| name | size | md5 | last_modified | creator |
| ---- | ---- | --- | ------------- | ------- |
| dt=2015/1680119600-25ee0299-28eb-47a4-b81e-f49f5399617b-0-4.parquet | 12616 | e273f913a5f75fea96f9fe7d2f086a6f | 2023-04-07 02:02:21.717 +0000 | NULL |
| dt=2015/1680119600-25ee0299-28eb-47a4-b81e-f49f5399617b-0-42.parquet | 5470 | a29c8427372d02dece62924dd985ae51 | 2023-04-07 02:02:21.725 +0000 | NULL |
| dt=2016/1680119600-25ee0299-28eb-47a4-b81e-f49f5399617b-37-26.parquet | 32708 | 2c21a9ffbec6d900f21fe4e8ff30d00f | 2023-04-07 02:02:22.131 +0000 | NULL |
| dt=2016/1680119600-25ee0299-28eb-47a4-b81e-f49f5399617b-37-27.parquet | 261112 | 8efed0a4490b52bb35a12eb1d2162d18 | 2023-04-07 02:02:21.724 +0000 | NULL |
| dt=2016/1680119600-25ee0299-28eb-47a4-b81e-f49f5399617b-37-44.parquet | 10927 | d5f7c980ac9e1e1c3de8d42c4388a7b1 | 2023-04-07 02:02:21.794 +0000 | NULL |
| dt=2016/1680119600-25ee0299-28eb-47a4-b81e-f49f5399617b-37-45.parquet | 273489 | e41f8268c73ff9e8062a16fd1bd3d8bf | 2023-04-07 02:02:22.180 +0000 | NULL |
| dt=2016/1680119600-25ee0299-28eb-47a4-b81e-f49f5399617b-37-9.parquet | 4220496 | f206d6bcb48483dc684ba8c67d5c6733 | 2023-04-07 02:02:22.145 +0000 | NULL |

> ### Creeate contract table

Reviewer: ```suggestion
### Create contract table
```

Build the target table using the following SQL

```sql
CREATE DATABASE indexedxyz;
USE indexedxyz;
CREATE TABLE `contract` (
  `block_time` BIGINT NULL,
  `address` VARCHAR NULL,
  `event_signature` VARCHAR NULL,
  `event_params` ARRAY(STRING NULL) NULL,
  `block_number` BIGINT NULL,
  `block_hash` VARCHAR NULL,
  `log_index` BIGINT NULL,
  `transaction_hash` VARCHAR NULL,
  `transaction_index` BIGINT NULL,
  `data` VARCHAR NULL,
  `topics` VARCHAR NULL,
  `id` VARCHAR NULL
);
```

> ### Load Data from stage

Reviewer: ```suggestion
### Load data from stage
```

```

| count() |
| 5374    |

And if we want to check on the transaction volume by month, use the following command

```sql
select date_trunc('month', to_timestamp(block_time)), count(*) from contract where lower(address) = '0xbc4ca0eda7647a8ab7c2061c2e118a18a936f13d' and event_signature = 'Transfer(address,address,uint256)' group by 1;
```

The result should be

| date_trunc(month, to_timestamp(block_time)) | count(*) |
| -------------------------------------------- | ------- |
| 2021-05-01 | 1742 |
| 2022-04-01 | 352 |
| 2023-03-01 | 1622 |
| 2022-07-01 | 244 |
| 2023-01-01 | 2556 |
| 2021-12-01 | 262 |
| 2021-10-01 | 190 |
| 2022-06-01 | 257 |
| 2022-02-01 | 205 |
| 2022-08-01 | 285 |
| 2021-07-01 | 357 |
| 2022-10-01 | 225 |
| 2021-11-01 | 219 |
| 2022-11-01 | 334 |
| 2021-08-01 | 563 |
| 2021-06-01 | 727 |
| 2023-04-01 | 121 |
| 2022-12-01 | 1566 |
| 2022-01-01 | 272 |
| 2022-05-01 | 426 |
| 2022-03-01 | 415 |
| 2021-04-01 | 48 |
| 2021-09-01 | 257 |
| 2023-02-01 | 1950 |
| 2022-09-01 | 245 |

> ### Visualize result through databend worksheet

Reviewer: ```suggestion
### Visualize result through Databend worksheet
```

> Databend Cloud natively provided a sort of visualization toolkits on worksheet.

Reviewer 2: how about clip the screen about how the visualization is configured ?

Reviewer 2: like this one: ![img_v2_1dc1cacd-cd2e-485e-86ce-408bc30fa50g](https://user-images.githubusercontent.com/129800/236435291-814833a6-7213-47dd-afc0-eb515d1f108b.jpg)

this image gives more infomation about how to configure a chart like this

Author: done
